### PR TITLE
Added function to calculate and encode relative operands

### PR DESCRIPTION
**Pull request by mappzor**
Related to [268](https://github.com/zyantific/zydis/issues/268#issuecomment-968218721) and #345.

This PR introduces `ZydisEncoderEncodeInstructionAbsolute` function which accepts absolute addresses instead of:
- RIP-relative operands (base is still EIP/RIP, displacement should contain absolute address) 
- relative immediates in branching instructions

First group of instructions is handled by patching displacement after encoding. Second group uses fast size prediction algorithm to find the right variant and calculate relative operand prior to encoding. This has less overhead than trying to brute-force encode different variants. Solutions exploits simplicity of branching instructions as they have less "moving parts" such as prefixes, ModR/M, SIB, etc. This allows for fast table lookups with some minor adjustments.

Let me know if you have any feedback. If we want to proceed with this solution I will clean up my test code and add it to this PR. Final test suite would be a small C tool triggered from usual python test scripts.

### CODE REVIEW

**Comment by mappzor** [comment]
Pushed some tests, CI fails because I cannot commit workflow change:
```diff
       - name: Running regression tests (encoder)
         run: |
           cd tests
-          python3 regression_encoder.py ../build/ZydisFuzzReEncoding ../build/ZydisFuzzEncoder
+          python3 regression_encoder.py ../build/ZydisFuzzReEncoding ../build/ZydisFuzzEncoder ../build/ZydisTestEncoderAbsolute
```

**Comment by flobernd** [comment]
Thanks a lot @mappzor 👍 

I just pushed the workflow changes. There seems to be a minor cast issue, but other than that it looks very good 🙂

**Review by flobernd — APPROVED** [review]
LGTM! Thanks again, that will make a lot of users happy!

Is there a python script to generate the new `.inc` file or was it rather "generated" by hand? 😄

**Comment by mappzor** [comment]
There's a new mode added to the script, will push it in a moment. It's not the prettiest thing but it does the job.

**Review by athre0z — APPROVED** [review]
LGTM as well, thanks a lot for tackling this!

File: src/Encoder.c [hunk 1]
```diff
@@ -4288,17 +4315,212 @@ ZYDIS_EXPORT ZyanStatus ZydisEncoderEncodeInstruction(const ZydisEncoderRequest
...
+                            forced_priority_row[0] = 2;
+                        }
+                        else
+                        {
+                            forced_priority_row[0] = (ZyanI8)(request->operand_size_hint - 1);
+                        }
+                        priority_row = (ZyanI8 *)&forced_priority_row;
+                    }
+                }
+                ZYAN_ASSERT(ZYAN_ARRAY_LENGTH(asz_priority[0]) ==
```
[the remark this answers is not in the document]
Comment (mappzor) [reply src/Encoder.c:4414]: Won't this trigger `unused-local-typedefs` warning in CI? I remember that used to be a thing in some configurations.

[the remark this answers is not in the document]
Comment (athre0z) [reply src/Encoder.c:4414]: We now require C11 on all compilers != MSVC, so the static assert should be a proper `_Static_assert` on all compilers that previously complained about that. But to be honest, I'm not 100% sure either and I don't feel like brute-forcing the CI, so let's just keep it as-is. :laughing:

**Pull request merged**